### PR TITLE
chore(doc-gen): fix the doc gen script

### PR DESCRIPTION
**Pull request by sjelin**
I never ran into the annotation problem @juliemr talked about, so please test this on your local machines

### CODE REVIEW

**Comment by juliemr** [comment]
I get `sh: gulp: command not found`. Confused as to where that's coming from - I'm looking into it...

File: scripts/generate-docs.sh [hunk 1]
```diff
@@ -27,8 +24,56 @@ if [ $? -ne 0 ]; then
...
+  echo -e "\033[0;31m" 1>&2 # Red
+  echo "Couldn't install testapp"
+  echo -e "\033[0m" 1>&2 # Normal Color
+  exit 1
+fi
 
-# Generate files
+echo "Building the website..."
+cd website
 npm run build
```
Comment (juliemr) [inline scripts/generate-docs.sh]: This should also run `npm install` here.

**Comment by juliemr** [comment]
I also get a lot of error messages but they don't seem to actually stop docs from being generated. 

Notably:

```
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.
```

I think this is due to old versions of some sub dependencies, probably from dgeni.

File: scripts/generate-docs.sh (hunk 1 above)
Comment (sjelin) [inline scripts/generate-docs.sh]: Sorry about that.  Looks like my clean from a couple lines up wasn't as thorough as I though, so I didn't catch this.  Fixed.

**Comment by sjelin** [comment]
@juliemr those definitely don't sound like a problem with the script itself.

**Comment by juliemr** [comment]
This looks pretty good to me and runs without completely falling over.

**Pull request merged**